File: src/main/java/com/wu/Hello.java
package com.wu;

//在IntelliJ 除了可使用VCS功能選單，也可按 Alt + 9 開啓版本控制視窗
//按 Alt + `(鍵盤左上同~鍵) 則可顯示常用版本控制(VC)動作

public class Hello {
    public static void main(String[] args) {  //Java程式的進入點，快捷鍵 psvm + Tab鍵
//        System.out.println("Hello world");  //快捷鍵 sout + Tab鍵
//        new Person().hello();  //用 new 生出Person物件，並使用其hello方法
        Person p = new Person();  //將 new 出的Person物件儲存於 p (Person物件)
        p.hello();  //呼叫 p (Person物件)身上的hello方法

        //Java的8種基本資料型態：
        int age = 19;  //32 bits 整數
        short num1 = 57;  //16 bits 整數
        long num2 = 1919;  //64 bits 整數
        byte num3 = 120;  //8 bits 整數
        char c = 'A';  //單一字元，也可儲存一個中文字
        float weight = 66.5f;  //Java會自動將有小數點數值設為double，所以需加上f告知為float
        double height = 1.7;
        boolean adult = true;  //布林值(true/false)
        //參照資料型態(屬於類別，所以首字大寫)，而且可使用功能(function)：
        String name = "Hank";  //字串(類別)
        //而上面的8種基本資料型態也都有相對應的類別(首字大寫)
        Integer age2 = 19;
        Character c2 = 'B';

    }
}

File: src/main/kotlin/com/wu/Hello.kt
package com.wu

//Kotlin的 main方法不需寄生在某個類別內部，可直接設在第一層(Top-level)
fun main(args: Array<String>) {  //快捷鍵 maina + Tab鍵。如不需接收執行時的參數，也可使用快捷鍵 main + Tab鍵，來省略小括號內的字串陣列
//    println("Hello Kotlin");  //直接呼叫Kotlin內建的 println()方法，Kotlin每行程式最後的分號(;)可省略
//    Human().hello()  //Kotlin呼叫類別不用new，可直接呼叫
    val h = Human()  //Kotlin使用 val(value不可再改變值)及 var(variable可改變值)來定義物件
        h.hello()

    //Kotlin沒有基本資料型態，只有類別(物件)的資料型態(首字大寫)
    //Kotlin的變數設定用 val(value，不可改變) 及 var(variable，可被改變)
    val age1 = 19;  //變數型態不用指定，可自動推斷
    val age2 : Int = 19;  //如要特別指定 val 名稱：型態 = 值
    var weight = 66.5;
    var name : String;
    name = "Hank";
}

class Human {  //增加Kotlin的類別不用另存成檔案
    fun hello() {  //Kotlin的方法如無回傳值則用不用寫
        println("Hello kotlin")
    }
}